File: src/Modules/Meetings/Domain/Members/Member.fs
namespace CompanyName.MyMeetings.Modules.Meetings.Domain.Members

open System
open CompanyName.MyMeetings.BuildingBlocks.Domain
open FsToolkit.ErrorHandling
//open CompanyName.MyMeetings.Modules.Meetings.Domain.Members.Events
open FsToolkit.ErrorHandling.Operator.Validation

type private MemberInternal = {
    MemberId: Guid
    Login: string
    FirstName: string
    LastName: string
    Name: string
    Email: string
    CreatedDate: DateTime
}

type Member = private Member of MemberInternal

type DomainResult = {
    Entity: Member
    Events: IDomainEvent list
}

module Member =
    
    let create id login firstname lastname name email =
        let f clock =
            let ev = Events.MemberCreatedDomainEvent(id)
            let m = Member ({MemberId = id; Login = login; FirstName = firstname; LastName = lastname; Name = name; Email = email; CreatedDate = clock})
            let e: IDomainEvent list = [ev]
            {Entity = m; Events = e}
        f

File: src/Modules/Meetings/Infrastructure/Database/Member.cs
using System;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace CompanyName.MyMeetings.Modules.Meetings.Infrastructure.Database
{
    public class Member
    {
        public Guid Id { get; set; }

        public string Login { get; set; }

        public string Email { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Name { get; set; }

        public DateTime CreatedDate { get; set; }
    }

    internal class MemberEntityTypeConfiguration : IEntityTypeConfiguration<Member>
    {
        public void Configure(EntityTypeBuilder<Member> builder)
        {
            builder.ToTable("Members", "meetings");
        }
    }
}